lib/portlayer/store/store.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"regexp"
+	"sync"
+
+	"github.com/vmware/vic/pkg/kvstore"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/vsphere/datastore"
+	"github.com/vmware/vic/pkg/vsphere/session"
+
+	"golang.org/x/net/context"
+)
+
+type StoreManager struct {
+	m            sync.RWMutex
+	dsStores     map[string]*kvstore.KeyValueStore
+	datastoreURL url.URL
+}
+
+var mgr *StoreManager
+
+const (
+	// default store folder name
+	dsFolder = "kvStores"
+	// available via portLayer API
+	APIKV = "apiKV"
+)
+
+var (
+	ErrDoesNotExit   = errors.New("requested store does not exist")
+	ErrDuplicateName = errors.New("duplicate store name")
+	ErrInvalidName   = errors.New("invalid store name, must be regexp(^[a-zA-Z0-9_]*$) compliant")
+)
+
+// Init will initialize the package vars and create the default portLayerKV store
+func Init(ctx context.Context, session *session.Session, imgStoreURL url.URL) error {
```

Comment:
Sure but should it be coupled like that? I think just make it generic -- its just a datastore.